services/shhext/chat/encryption_test.go

```diff
@@ -314,6 +314,171 @@ func (s *EncryptionServiceTestSuite) TestConversation() {
 	s.Equal(cleartext2, decryptedPayload1, "It correctly decrypts the payload using X3DH")
 }
 
+// Previous implementation allowed max maxSkip keys in the same receiving chain
+// leading to a problem whereby dropped messages would accumulate and eventually
+// we would not be able to decrypt any new message anymore.
+// Here we are testing that maxSkip only applies to *consecutive* messages, not
+// overall.
+func (s *EncryptionServiceTestSuite) TestMaxSkipKeys() {
+	bobText := []byte("text")
+
+	bobKey, err := crypto.GenerateKey()
+	s.Require().NoError(err)
+
+	aliceKey, err := crypto.GenerateKey()
+	s.Require().NoError(err)
+
+	// Create a bundle
+	bobBundle, err := s.bob.CreateBundle(bobKey)
+	s.Require().NoError(err)
+
+	// We add bob bundle
+	_, err = s.alice.ProcessPublicBundle(aliceKey, bobBundle)
+	s.Require().NoError(err)
+
+	// Create a bundle
+	aliceBundle, err := s.alice.CreateBundle(aliceKey)
+	s.Require().NoError(err)
+
+	// We add alice bundle
+	_, err = s.bob.ProcessPublicBundle(bobKey, aliceBundle)
+	s.Require().NoError(err)
+
+	// Bob sends a message
+
+	for i := 0; i < maxSkip; i++ {
+		_, err = s.bob.EncryptPayload(&aliceKey.PublicKey, bobKey, bobText)
+		s.Require().NoError(err)
+	}
+
+	// Bob sends a message
+	bobMessage1, err := s.bob.EncryptPayload(&aliceKey.PublicKey, bobKey, bobText)
+	s.Require().NoError(err)
+
+	// Alice receives the message
+	_, err = s.alice.DecryptPayload(aliceKey, &bobKey.PublicKey, bobInstallationID, bobMessage1)
+	s.Require().NoError(err)
+
+	// Bob sends a message
+	_, err = s.bob.EncryptPayload(&aliceKey.PublicKey, bobKey, bobText)
+	s.Require().NoError(err)
+
+	// Bob sends a message
+	bobMessage2, err := s.bob.EncryptPayload(&aliceKey.PublicKey, bobKey, bobText)
+	s.Require().NoError(err)
+
+	// Alice receives the message, we should have 1001 keys in the db, but
+	// we should not throw an error
+	_, err = s.alice.DecryptPayload(aliceKey, &bobKey.PublicKey, bobInstallationID, bobMessage2)
```

Comment:
In which situation, is an error returned? Should we cover it with a unit test?

Reply:
basically if peer A sends maxSkip messages to peer B, but B never receives them. If peer A sends another message when peer B receives it, it will throw an error.
I will add a test, I somehow missed that

Reply:
added test!